### PR TITLE
Use fp32 inference precision

### PR DESCRIPTION
**Pull request by andrey-churkin**
### Changes

- Use fp32 inference precision

### Reason for changes

Ref: 140438

### Related tickets

Ref: 140438

### CODE REVIEW

**Comment by andrey-churkin** [comment]
@alexsu52 @KodiaqQ Are you guys okay with how this was implemented, or are any changes required? Please review, as I will start validation soon.

File: nncf/openvino/engine.py [hunk 1]
```diff
@@ -62,10 +64,13 @@ class OVNativeEngine(Engine):
     to infer the model.
     """
 
-    def __init__(self, model: ov.Model):
+    def __init__(self, model: ov.Model, use_fp32_precision: bool = False):
```
Comment (alexsu52) [inline nncf/openvino/engine.py:67]:
```suggestion
    def __init__(self, model: ov.Model, use_fp32_precision: bool = True):
```
Is it possible?

Comment (andrey-churkin) [reply nncf/openvino/engine.py:67]: Yes, it allows us to get the same results for SPR and VNNI. My question is: do we really need such a parameter, or is it better to use inference_precision: Type.f32 for all cases?

Comment (andrey-churkin) [reply nncf/openvino/engine.py]: Done

**Comment by andrey-churkin** [comment]
Should I use the fp32 hint here as well?

https://github.com/openvinotoolkit/nncf/blob/dfc5d78f76406403ae25c3bbbc7c5881cd8837f8/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L348

https://github.com/openvinotoolkit/nncf/blob/dfc5d78f76406403ae25c3bbbc7c5881cd8837f8/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L381

**Comment by andrey-churkin** [comment]
@alexsu52 @KodiaqQ Please review

**Comment by alexsu52** [comment]
> Should I use the fp32 hint here as well?
> 
> https://github.com/openvinotoolkit/nncf/blob/dfc5d78f76406403ae25c3bbbc7c5881cd8837f8/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L348
> 
> https://github.com/openvinotoolkit/nncf/blob/dfc5d78f76406403ae25c3bbbc7c5881cd8837f8/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L381

Yes, you should use the fp32 hint there as well.

**Review by alexsu52 — APPROVED** [review]
LGTM. Please provide ptq conformance test results.

**Review by KodiaqQ — APPROVED** [review]

**Comment by andrey-churkin** [comment]
NNCF/manual/post_training_quantization: (develop) Build # 517
 NNCF/manual/post_training_quantization: Build # 516

**Pull request merged**